distributed/executor.py

```diff
@@ -987,58 +987,101 @@ def cancel(self, futures):
         return sync(self.loop, self._cancel, futures)
 
     @gen.coroutine
-    def _publish_dataset(self, data, name):
-        keys = [tokey(f.key) for f in futures_of(data)]
-        out = yield self.scheduler.publish_dataset(keys=keys, name=tokey(name),
-                                             data=dumps(data),
-                                             client=self.id)
-        if out['status'] != "OK":
-            raise KeyError(name)
-
-    def publish_dataset(self, data, name):
+    def _publish_dataset(self, **kwargs):
+        coroutines = []
+        for name, data in kwargs.items():
+            keys = [tokey(f.key) for f in futures_of(data)]
+            coroutines.append(self.scheduler.publish_dataset(keys=keys,
+                name=tokey(name), data=dumps(data), client=self.id))
```

Comment:
How about only allow two cases: is a sequence where type of every item is future, or has .dask attribute. Easy to spoof, of course.